attestation/attestation.go

```diff
@@ -141,10 +140,52 @@ func CreateAIK(context *tspi.Context) ([]byte, []byte, error) {
 	return pubkey, blob, nil
 }
 
+// GetQuote consumes a nonce and the aik public blob and returns the quote,
+// a signature, and any error
+func GetQuote(context *tspi.Context, aikblob, nonce []byte) ([]byte, []byte, error) {
+	srk, err := context.LoadKeyByUUID(tspiconst.TSS_PS_TYPE_SYSTEM, tspi.TSS_UUID_SRK)
+	if err != nil {
+		return nil, nil, err
+	}
+	srkpolicy, err := srk.GetPolicy(tspiconst.TSS_POLICY_USAGE)
+	if err != nil {
+		return nil, nil, err
+	}
+	srkpolicy.SetSecret(tspiconst.TSS_SECRET_MODE_SHA1, wellKnown[:])
+
+	tpm := context.GetTPM()
+	tpmpolicy, err := context.CreatePolicy(tspiconst.TSS_POLICY_USAGE)
+	if err != nil {
+		return nil, nil, err
+	}
+	tpm.AssignPolicy(tpmpolicy)
+	tpmpolicy.SetSecret(tspiconst.TSS_SECRET_MODE_SHA1, wellKnown[:])
+
+	aik, err := context.LoadKeyByBlob(srk, aikblob)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	pcrs, err := context.CreatePCRs(tspiconst.TSS_PCRS_STRUCT_DEFAULT)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to get a reference to PCRs: %v", err)
+	}
+
+	bitmap := make([]int, 24)
```

Comment:
Why is this created but not returned? Please add a comment and explain what the values you're setting it to.

Reply:
I suppose it's a bit inaccurate to call it a bitmap. SetPCRs consumes a slice of chosen PCRs (when we later get the PCR values it will only fetch values of PCRs in the slice). I would much prefer to do something like pcrs.SetPCRS(range(24)) but I don't believe there is anyway to do so in go?

Reply:
Try this:

```
pcrValues := make([]int, 24)
for i := range pcrValues {
    pcrValues[i] = i
}
if err := pcrs.SetPCRs(pcrValues); err != nil {
    // ...
}
```

Also please be more descriptive in your comment. It's clear that you're setting all PCR values, but it's not clear _why_ you're setting them or _why_ you're setting the to this value.

Reply:
I think it might be better to have that as a comment at the function signature.
We get all of the PCRs just so that we don't miss any data. Not all of them may have been used but the cost benefits to only getting some of the PCRs is kind of negligible.
Also I'm not sure if maybe this is a misunderstanding (because the naming is kind of terrible) but we're not setting the PCR values, were just setting a mask (the mask just isn't in the form of selected/not it's stored as a list of selected PCRs hence why we set the mask to range(24)).